include/deal.II/base/parameter_acceptor.h

```diff
@@ -0,0 +1,528 @@
+ *
+ * If you pass an empty name, the boost::core::demangle() function is used to
+ * fill the section name with a human readable version of the class name
+ * itself.
+ *
+ * @author Luca Heltai, 2017.
+ */
+class ParameterAcceptor : public Subscriptor
+{
+public:
+  /**
+   * The constructor adds derived classes to the list of acceptors. If
+   * a section name is specified, then this is used to scope the
+   * parameters in the given section, otherwise a pretty printed
+   * version of the derived class is used.
+   */
+  ParameterAcceptor(const std::string section_name="");
+
+  /**
+   * The destructor sets to zero the pointer relative to this index,
+   * so that it is safe to destroy the mother class.
+   */
+  virtual ~ParameterAcceptor();
+
+  /**
+   * Call declare_all_parameters(), read the parameters from `filename` (only
+   * if `filename` is a non-empty string), and then call
+   * parse_all_parameters().
+   *
+   * If the parameter `filename` is the empty string, then no attempt to read a
+   * parameter file is done. This may be useful if you are ok with using
+   * default values, and don't want to read external files to use a class
+   * derived from ParameterAcceptor.
+   *
+   * If outfilename is not the emtpy string, then write the content that was
+   * read in to the outfilename. The format of both input and output files are
+   * selected using the extensions of the files themselves. This can be either
+   * `prm` or `xml` for input, and `prm`, `xml`, or `tex/latex` for output. If
+   * the output format is `prm`, then `output_style_for_prm_format` is used to
+   * decide wether we write the full documentation as well, or only the
+   * parameters.
+   *
+   * If the input file does not exist, a default one with the same name is created
+   * for you, and an exception is thrown.
+   *
+   * @param filename Input file name
+   * @param output_filename Output file name
+   * @param output_style_for_prm_format How to write the output file if format is `prm`
+   */
+  static void initialize(const std::string filename="",
+                         const std::string output_filename="",
+                         const ParameterHandler::OutputStyle output_style_for_prm_format=ParameterHandler::ShortText);
+
+  /**
+   * Clear class list and global parameter file.
+   */
+  static void clear();
+
+  /**
+   * Declare parameter entries of the derived class.
+   */
+  virtual void declare_parameters(ParameterHandler &prm);
+
+  /**
+   * Declare parameter call back. This function is called at the end of
+   * declare_all_parameters, to allow users to process their parameters right
+   * after they have been parsed. The default implementation is empty.
+   *
+   * You can use this function, for example, to create a quadrature
+   * rule after you have read how many quadrature points you wanted
+   * to use from the parameter file.
+   */
+  boost::signals2::signal<void()> declare_parameters_call_back;
```

Comment:
but if this is called after declaring parameters, you don't know how many quadrature points you have. Is this a copy-paste error from the other signal?